File: app/src/main/kotlin/com/arcao/opencaching4locus/data/network/interceptor/AuthenticationInterceptor.java
package com.arcao.opencaching4locus.data.network.interceptor;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthenticationInterceptor implements Interceptor {

    private String authToken;

    public AuthenticationInterceptor(String token) {
        this.authToken = token;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request original = chain.request();

        Request request = original
                .newBuilder()
                .header("Authorization", authToken)
                .build();

        return chain.proceed(request);
    }
}

File: app/src/main/kotlin/com/arcao/opencaching4locus/data/network/interceptor/AuthenticationInterceptor.kt
package com.arcao.opencaching4locus.data.network.interceptor

import android.content.Context
import com.arcao.opencaching4locus.data.network.OkApiServiceType
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class AuthenticationInterceptor(private val context: Context, serviceType: OkApiServiceType) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val original = chain.request()

        val request = original
                .newBuilder()
                .header("Authorization", "")
                .build()

        return chain.proceed(request)
    }
}